core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/PathTrackingInputFormat.java

```diff
@@ -70,14 +82,23 @@ public static Schema getOutputSchema(@Nullable String pathField) {
       // should never happen
       throw new IllegalStateException("Input split is not a FileSplit.");
     }
-
-    RecordReader<LongWritable, Text> delegate = (new TextInputFormat()).createRecordReader(split, context);
     FileSplit fileSplit = (FileSplit) split;
     String pathField = context.getConfiguration().get(PATH_FIELD);
     boolean filenameOnly = context.getConfiguration().getBoolean(FILENAME_ONLY, false);
     String path = filenameOnly ? fileSplit.getPath().getName() : fileSplit.getPath().toUri().toString();
-
-    return new TrackingTextRecordReader(delegate, pathField, path);
+    fileSourceFormat = fileSourceFormat.toLowerCase();
```

Comment:
you should be reading the format from the configuration, similar to how the path field and filenameOnly are being read.